sdcm/provision/azure/utils.py

```diff
@@ -67,8 +68,16 @@ def get_scylla_images(  # pylint: disable=too-many-branches
                     break
             else:
                 output.append(image)
+    output.sort(key=lambda img: int(img.tags.get('build_id', "0")))
+    try:
+        # example image name: ScyllaDB-5.1.dev-0.20220603.72f629c2b6e8-1-build-157
+        output.sort(key=lambda img: int(SCYLLA_VERSION_GROUPED_RE.match(
+            img.name.split("ScyllaDB-", 1)[-1]).group("date")))
+    except AttributeError as exc:
+        unparsable_scylla_versions = [img.name for img in output
+                                      if not SCYLLA_VERSION_GROUPED_RE.match(img.name.split("ScyllaDB-", 1)[-1])]
+        raise AttributeError(f"Cannot parse ScyllaVersions: {unparsable_scylla_versions}") from exc
```

Comment:
2 problems with it:

1) Are you sure you want to raise exception if sorting cannot be done precisely?
2) If there appears some image with the name that is not parsable then all the runs will fail on this exception. And this situation will become a real problem.
Maybe we just should walk through all the versions, exclude unparsable ones and then use only proper ones for results. If there are unparsable ones - raise `warning` or `error` SCT event.

Reply:
I adapted the code and added unit test for it. I decided to use logging instead of sct event as it's not usual to send warning/error events for such things.